python/ray/scripts/scripts.py

```diff
@@ -1123,13 +1123,14 @@ def on_terminate(proc):
     grace_period_to_kill_gcs = int(grace_period / 2)
     grace_period_to_kill_components = grace_period - grace_period_to_kill_gcs
 
-    # Kill evertyhing except GCS.
-    found, stopped, alive = kill_procs(
-        force, grace_period_to_kill_components, processes_to_kill[1:]
-    )
-    total_procs_found += found
-    total_procs_stopped += stopped
-    procs_not_gracefully_killed.extend(alive)
+    # Kill evertyhing except GCS, one kind at a time.
+    for process in processes_to_kill[1:]:
+        found, stopped, alive = kill_procs(
+            force, grace_period_to_kill_components, [process]
+        )
+        total_procs_found += found
+        total_procs_stopped += stopped
+        procs_not_gracefully_killed.extend(alive)
```

Comment:
If we want to stop the process in a deterministic order, is it a good idea to inject some waiting time at the end of each loop?

Reply:
we already do terminate-wait-kill-wait in `kill_proc`: https://github.com/ray-project/ray/blob/8120b6ac86f31d95ae47d93acfd55d54e5dec0fc/python/ray/scripts/scripts.py#L1113